modules/ballerina-core/src/main/java/org/ballerinalang/model/values/BArray.java

```diff
@@ -51,24 +53,34 @@ public BArray(Class<V> valueClass) {
         this.valueClass = valueClass;
     }
 
-    public <V extends BValue> void add(int index, V value) {
-        ensureCapacity(index);
+    public <V extends BValue> void add(long index, V value) {
+        if (index != (int) index) {
```

Comment:
I did run a simple perf test and according to that, there is a nanosecond perf gain when using index < Integer.MAX_VALUE logic. Hence changed the logic to that